### ISSUE TITLE
Allow non-block themes with template parts support to create custom template parts in the Site Editor

### ISSUE THREAD

**Issue by aurooba** [WordPress/gutenberg#51099]
## What problem does this address?

Right now, if your theme is not a block theme, you don't have the ability to create `general` template parts from the Site Editor. The only template parts you have access to are the ones added in your theme. It would be very helpful from a content administration perspective to be able to create custom template parts that could be used in other places.

## What is your proposed solution?

I could see this being solved by the upcoming `Library` esque feature combined with more powerful patterns that you can create/edit without diving into the code, as well. But I do think the ability to create `general` template parts in the site editor should be an opt-in setting for themes. It would solve quite a few workflow issues.

**Comment by t-hamano** [comment]
My understanding is that in non-blocking themes (classic themes with theme.json), the inability to create template parts in the Site Editor is intentional.

Even if we are able to create a new template parts in the Site Editor, it must be injected into a PHP-based theme template file in order to be displayed on the front end. Furthermore, since the slug for template parts created in the Site Editor is automatically generated, many people are confused about what slug to specify as the argument to the `block_template_part()` function.

In other words, we cannot completely manage template parts only in the Site Editor, as we can in the block editor. I think this is the reason why non-blocking themes cannot create template parts in the Site Editor.

@carolinan

Do you know anything about this issue?

**Comment by jordesign** [comment]
With Synced Patterns being introduced - is there any reason that can't be used instead of introducing the ability to add a Template Part?

**Comment by t-hamano** [comment]
I just wanted to check the status of this issue. 

The latest version of Gutenberg allows you to create sunced/unsynced patterns and also assign pattern categories. My understanding is that it is intentional that non-block-based themes do not allow the creation of template parts.

By the way, I would like to mention that there is also #52150, which aims to allow non-block-based themes to access the Site Editor's pattern UI.

**Comment by t-hamano** [comment]
As mentioned in [this comment](https://github.com/WordPress/gutenberg/issues/51099#issuecomment-1585675289), I think this is the intended design. I'd like to close this issue, but if there's anything I've missed, please feel free to comment.
